tests/staggered_eigensolve_test.cpp

```diff
@@ -57,372 +33,178 @@ void display_test_info()
-
-  // FIXME: currently assume staggered is SU(3)
-  gauge_param.type = (dslash_type == QUDA_STAGGERED_DSLASH || dslash_type == QUDA_LAPLACE_DSLASH) ?
-    QUDA_SU3_LINKS :
-    QUDA_ASQTAD_FAT_LINKS;
-  gauge_param.reconstruct = QUDA_RECONSTRUCT_NO;
-  GaugeFieldParam cpuFatParam(milc_fatlink, gauge_param);
-  cpuFatParam.ghostExchange = QUDA_GHOST_EXCHANGE_PAD;
-  cpuGaugeField *cpuFat = new cpuGaugeField(cpuFatParam);
-  ghost_fatlink = (void **)cpuFat->Ghost();
-
-  gauge_param.type = QUDA_ASQTAD_LONG_LINKS;
-  GaugeFieldParam cpuLongParam(milc_longlink, gauge_param);
-  cpuLongParam.ghostExchange = QUDA_GHOST_EXCHANGE_PAD;
-  cpuGaugeField *cpuLong = new cpuGaugeField(cpuLongParam);
-  ghost_longlink = (void **)cpuLong->Ghost();
-
-#else
-  int fat_pad = 0;
-  int link_pad = 0;
-#endif
-
-  gauge_param.type = (dslash_type == QUDA_STAGGERED_DSLASH || dslash_type == QUDA_LAPLACE_DSLASH) ?
-    QUDA_SU3_LINKS :
-    QUDA_ASQTAD_FAT_LINKS;
-  gauge_param.ga_pad = fat_pad;
-  if (dslash_type == QUDA_STAGGERED_DSLASH || dslash_type == QUDA_LAPLACE_DSLASH) {
-    gauge_param.reconstruct = link_recon;
-    gauge_param.reconstruct_sloppy = link_recon_sloppy;
-  } else {
-    gauge_param.reconstruct = gauge_param.reconstruct_sloppy = QUDA_RECONSTRUCT_NO;
-  }
-  gauge_param.cuda_prec_precondition = gauge_param.cuda_prec_sloppy;
-  gauge_param.reconstruct_precondition = gauge_param.reconstruct_sloppy;
-  loadGaugeQuda(milc_fatlink, &gauge_param);
+  // Reorder gauge fields to MILC order
+  reorderQDPtoMILC(milc_fatlink, qdp_fatlink, V, gauge_site_size, gauge_param.cpu_prec, gauge_param.cpu_prec);
+  reorderQDPtoMILC(milc_longlink, qdp_longlink, V, gauge_site_size, gauge_param.cpu_prec, gauge_param.cpu_prec);
 
-  if (dslash_type == QUDA_ASQTAD_DSLASH) {
-    gauge_param.type = QUDA_ASQTAD_LONG_LINKS;
-    gauge_param.ga_pad = link_pad;
-    gauge_param.staggered_phase_type = QUDA_STAGGERED_PHASE_NO;
-    gauge_param.reconstruct = link_recon;
-    gauge_param.reconstruct_sloppy = link_recon_sloppy;
-    gauge_param.cuda_prec_precondition = gauge_param.cuda_prec_sloppy;
-    gauge_param.reconstruct_precondition = gauge_param.reconstruct_sloppy;
-    loadGaugeQuda(milc_longlink, &gauge_param);
+  loadFatLongGaugeQuda(milc_fatlink, milc_longlink, gauge_param);
+
+  // Staggered Gauge construct END
+  //-----------------------------------------------------------------------------------
+
+  // Host side arrays to store the eigenpairs computed by QUDA
+  void **host_evecs = (void **)malloc(eig_nConv * sizeof(void *));
+  for (int i = 0; i < eig_nConv; i++) {
+    host_evecs[i] = (void *)malloc(V * my_spinor_site_size * eig_inv_param.cpu_prec);
   }
+  double _Complex *host_evals = (double _Complex *)malloc(eig_param.nEv * sizeof(double _Complex));
 
+  double time = 0.0;
+
+  // QUDA eigensolver test
+  //----------------------------------------------------------------------------
   switch (test_type) {
   case 0: // full parity solution
```

Comment:
this is already caught above

Reply:
https://github.com/lattice/quda/commit/77a672f8699d8bbc4a0a9c7614bbbee008263f91